csharp/src/SeedLang/Runtime/HeapObject.cs

```diff
@@ -0,0 +1,84 @@
+// Copyright 2021 The Aha001 Team.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+
+namespace SeedLang.Runtime {
+  internal class HeapObject {
+    private readonly object _object;
+
+    private HeapObject(string str) {
+      _object = str;
+    }
+
+    private HeapObject(List<Value> list) {
+      _object = list;
+    }
+
+    internal static HeapObject String(string str) {
+      return new HeapObject(str);
+    }
+
+    internal static HeapObject List(List<Value> values) {
+      return new HeapObject(values);
+    }
+
+    internal string AsString() {
+      switch (_object) {
+        case string str:
+          return str;
+        case List<Value> list:
+          return list.ToString();
+        default:
+          throw new NotImplementedException($"Unsupported box type: {_object}");
```

Comment:
"box type", or "heap object type"?

Here and the rest.

Reply:
Done